### PR TITLE
Add Mesh2dInsertEdgeFromCoordinates

### PR DESCRIPTION
**Pull request by lucacarniato**

### CODE REVIEW

File: src/MeshKernelNET/Api/IMeshKernelApi.cs [hunk 1]
```diff
@@ -1040,6 +1040,25 @@ public interface IMeshKernelApi : IDisposable
         /// <returns>Error code</returns>
         int Mesh2dInsertEdge(int meshKernelId, int startVertexIndex, int endVertexIndex, ref int edgeIndex);
 
+        /// @brief Insert a new mesh2d edge from 2 coordinates
+        /// <param name="meshKernelId"> The id of the mesh state
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Api/IMeshKernelApi.cs:1044]: Add an explanation that if the coordinate does not (exactly? within some radius?) match a vertex, then a new one is created.

File: test/MeshKernelNETTest/Api/MeshKernelTest.cs [hunk 2]
```diff
@@ -156,6 +156,67 @@ public void Mesh2dInsertEdgeThroughApi()
             }
         }
 
+        [Test]
+        public void Mesh2dInsertEdgeFromCoordinatesThroughApi()
```
Comment (andreasbuykx) [inline test/MeshKernelNETTest/Api/MeshKernelTest.cs:160]: Would be nice to also have a test with inserting an edge with only one coordinate matching an existing vertex, and one with no coordinate matching an existing vertex.

**Review by andreasbuykx — APPROVED** [review]
